pkgs/misc/vscode-extensions/rust-analyzer/default.nix

```diff
@@ -0,0 +1,50 @@
+# Update script: pkgs/development/tools/rust/rust-analyzer/update.sh
+{ lib, pkgs, stdenv, nodejs, vscode-utils, jq, rust-analyzer
+, setDefaultRaLspServerPath ? true
+}:
+
+let
+  pname = "rust-analyzer";
+  publisher = "matklad";
+
+  # Follow the unstable version of rust-analyzer, since the extension is not stable yet.
+  inherit (rust-analyzer) version;
+
+  vsix = (import ./node-composition.nix {
```

Comment:
Can you move this to /home/joerg/git/nixpkgs/pkgs/development/node-packages/node-packages-v10.json ?

Reply:
Your update script could also update the version in that json file.

Reply:
@Mic92 
I saw the syntax is like `{"npm2nix": "git://github.com/NixOS/npm2nix.git#5.12.0"}`.  But is there any way to specify **a sub-directory** of github repository as the package root in `node-packages-v*.json`? The `package.json` is [under `editors/code`](https://github.com/rust-analyzer/rust-analyzer/tree/master/editors/code).

Reply:
It is not supported. It might be possible to implement this in future, but the current solution looks good to me!